spec/03-plugins/28-grpc-gateway/01-proxy_spec.lua

```diff
@@ -153,42 +160,1151 @@ for _, strategy in helpers.each_strategy() do
+                  },
+                  stringMap = {
+                    ["200"] = { any = { ["@type"] = "type.googleapis.com/targetservice.ComplexType", enumVal = 2 } },
+                  }
+                }
+              }
+            }
+          }
+        }
+        local res, _ = proxy_client:post( "/bounceMaskedFields", {
+          headers = { ["Content-Type"] = "application/json" },
+          body = request
+        })
+
+        assert.equal(200, res.status)
+        assert.same({
+          complexValue = {
+            anyProcessed = true, boolVal = true,
+            any = {
+              ["@type"] = "type.googleapis.com/targetservice.ComplexType", 
+              repeatedComplex = {
+                { anyProcessed = true, any = { ["@type"] = "type.googleapis.com/targetservice.ComplexType", enumVal = 1, boolVal = true }, boolVal = true },
+                { anyProcessed = true, any = { ["@type"] = "type.googleapis.com/targetservice.ComplexType", enumVal = 2, boolVal = true }, boolVal = true }
+              },
+              anyProcessed = true, boolVal = true,
+              any = { ["@type"] = "type.googleapis.com/targetservice.ComplexType", boolVal = true,
+                singularComplex = {
+                  boolVal = true,
+                  intMap = {
+                    ["200"] = { anyProcessed = true, any = { ["@type"] = "type.googleapis.com/targetservice.ComplexType", enumVal = 1, boolVal = true }, boolVal = true },
+                  },
+                  stringMap = {
+                    ["200"] = { anyProcessed = true, any = { ["@type"] = "type.googleapis.com/targetservice.ComplexType", enumVal = 2, boolVal = true }, boolVal = true },
+                  }
+                } 
+              }
+            }
+          }
+        }, cjson.decode( (res:read_body()) ))
+      end)
+    end)
+
+    it("invalid bool value", function()
+      local res, _ = proxy_client:post( "/bounceMaskedFields", {
+        headers = { ["Content-Type"] = "application/json" },
+        body = { complexValue = { bool_val = "untrue" } }
+      })
+      
+      assert.equal(400, res.status)
+      assert.equal("expected boolean value at .complexValue.bool_val, got: 'untrue' of type: string", (res:read_body()))
+    end)
+  
     test("structured URI args", function()
       local res, _ = proxy_client:get("/v1/grow/tail", {
         query = {
           name = "lizard",
           hands = { count = 0, endings = "fingers" },
           legs = { count = 4, endings = "toes" },
-          tail = {count = 0, endings = "tip" },
+          tail = { count = 0, endings = "tip" },
         }
       })
       assert.equal(200, res.status)
       local body = assert(res:read_body())
       assert.same({
         name = "lizard",
-        hands = { count = 0, endings = "fingers" },
+        hands = { endings = "fingers" }, -- count = 0 is default value, skipped from result
```

Comment:
We should not change this behavior. Maybe we should make the `default handling` setting by default to emit the fields.

Reply:
fixed in https://github.com/Kong/kong/pull/8297/commits/891d1f010adc766c92d0ec438be157afc5330445